internal/management/controller/tablespaces/infrastructure/postgres.go

```diff
@@ -0,0 +1,193 @@
+			&tbs.Name,
+			&tbs.Temporary,
+		)
+		if err != nil {
+			return nil, wrapErr(err)
+		}
+		tablespaces = append(tablespaces, tbs)
+	}
+
+	if rows.Err() != nil {
+		return nil, wrapErr(rows.Err())
+	}
+	return tablespaces, nil
+}
+
+// Update the tablespace in the database
+// we only allow the tablespace update the temporary attribute for tablespace right now
+func (tbsMgr postgresTablespaceManager) Update(ctx context.Context, tbs Tablespace) error {
+	contextLog := log.FromContext(ctx).WithName("tbs_reconciler_update")
+	contextLog.Trace("Invoked Update", "tbs", tbs)
+	wrapErr := func(err error) error {
+		return fmt.Errorf("while update tablespace %s: %w", tbs.Name, err)
+	}
+
+	tempTbs, err := tbsMgr.getCurrentTemporaryTablespaces(ctx)
+	if err != nil {
+		return wrapErr(err)
+	}
+
+	idx := slices.Index(tempTbs, tbs.Name)
+
+	var needUpdate bool
+	if tbs.Temporary && idx < 0 {
+		tempTbs = append(tempTbs, tbs.Name)
+		needUpdate = true
+	}
+
+	if !tbs.Temporary && idx >= 0 {
+		tempTbs = append(tempTbs[:idx], tempTbs[idx+1:]...)
+		needUpdate = true
+	}
+
+	if needUpdate {
+		contextLog.Debug("Update tablespace to temporary", "tbs", tbs.Name, "temporary", strings.Join(tempTbs, ","))
+		if _, err = tbsMgr.superUserDB.ExecContext(ctx, fmt.Sprintf("ALTER SYSTEM SET %s = %s",
+			sysVarTmpTbs,
+			strings.Join(tempTbs, ","),
+		)); err != nil {
+			return wrapErr(err)
+		}
+		if _, err = tbsMgr.superUserDB.ExecContext(ctx, "SELECT pg_reload_conf()"); err != nil {
+			return wrapErr(err)
+		}
+	}
+
+	return nil
+}
+
+// Create the tablespace in the database, if tablespace is temporary tablespace, need reload configure
+func (tbsMgr postgresTablespaceManager) Create(ctx context.Context, tbs Tablespace) error {
+	contextLog := log.FromContext(ctx).WithName("tbs_reconciler_create")
+	contextLog.Trace("Invoked Create", "tbs", tbs)
+	wrapErr := func(err error) error {
+		return fmt.Errorf("while creating tablespace %s: %w", tbs.Name, err)
+	}
+	var err error
+	if _, err = tbsMgr.superUserDB.ExecContext(ctx, fmt.Sprintf("CREATE TABLESPACE %s LOCATION '%s'",
+		pgx.Identifier{tbs.Name}.Sanitize(),
+		specs.LocationForTablespace(tbs.Name),
+	)); err != nil {
+		return wrapErr(err)
+	}
+
+	var tempTbs []string
+	var needUpdate bool
+	if tbs.Temporary {
+		if tempTbs, err = tbsMgr.getCurrentTemporaryTablespaces(ctx); err != nil {
+			return wrapErr(err)
+		}
+		idx := slices.Index(tempTbs, tbs.Name)
+		if idx < 0 {
+			tempTbs = append(tempTbs, tbs.Name)
+			needUpdate = true
+		}
+	}
+
+	if needUpdate {
+		contextLog.Debug("Set tablespace to template", "tbs", tbs.Name, "temporary value", strings.Join(tempTbs, ","))
+		if _, err = tbsMgr.superUserDB.ExecContext(ctx, fmt.Sprintf("ALTER SYSTEM SET %s = %s",
+			sysVarTmpTbs,
+			strings.Join(tempTbs, ","),
+		)); err != nil {
+			return wrapErr(err)
+		}
+		if _, err = tbsMgr.superUserDB.ExecContext(ctx, "SELECT pg_reload_conf()"); err != nil {
+			return wrapErr(err)
+		}
+	}
+
+	return nil
+}
```

Comment:
I think we should try to avoid alter system. 
We could try to add the tablespace names when we generate the conf?

Reply:
After 4ec09e6c4a969f870944ace682986b3f34c678a3 we cannot use ALTER SYSTEM.
We need to put the parameter directly in the custom.conf file.
The temporary tablespace setting can be updated even when the tablespace has not yet been created.

Reply:
In both the uses of ALTER SYSTEM, the functionality was to set the tablespaces to TEMPORARY.
That functionality was not yet expected of the Declarative Tablespaces feature.
We should remove all that.
It was left to get Marco's comments ... back in early September.
cc/ @leonardoce as you have this in review...

Reply:
We should never use `ALTER SYSTEM` to control the configuration of Postgres. This is not related at all with #2812 or the future work towareds disabling `ALTER SYSTEM` altogether.

Reply:
done